### ISSUE TITLE
ipfs:// and seamless upgrade of CIDv0

### ISSUE THREAD

**Issue by lidel** [eversum/galacteek#5]
Hi, something I noticed whem playing with protocol handlers: when user tries to open
```
ipfs://QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR
```
address bar breaks case-sensitive CIDv0 by forcing lowercase on it:

```
ipfs://qmbwqxbekc3p8tqskc98xmwnzrzdtrlmimpl8wbutgsmnr/
```

Would it be possible to convert cidv0 to case-insensitive cidv1b32 on the fly?

**Comment by eversum** [comment]
Hi @lidel !

Yes it's true that Qt will do that, because the ipfs:// URL scheme is declared in galacteek as having the 'Host' syntax, and i believe that implies that the host part of the URL needs to be lowercase (and Qt forces it itself if it's not already lowecase).

That's a great suggestion though i will have it implemented for the next version, thanks ! Right now in the URL bar you can put a simple CID or path (without any scheme) and it will be loaded with the best fitting scheme but it doesn't do CID conversion on the fly yet.

I'm working on various things related to URL schemes that will be quite handy to have i think.

Thanks.

**Comment by eversum** [comment]
The automatic CID upgrade has been implemented in version 0.4.12 and it's enabled in most places, so it's a bit of a base32 festival ... Hopefully it doesn't break too many things :| One goal is to have all hashmarks to be recorded using base32 encoded CIDS so that they're really unique.

Also v0.4.12 brings the possibility to pin IPFS links found in a webpage, and possibility to map IPFS objects for quick access using the q:// URL scheme.

Thanks again for the suggestion.

**Comment by lidel** [comment]
Defaulting to CIDv1 in BAse32 is a good idea (IPFS will switch to it as a default at some point: https://github.com/ipfs/ipfs/issues/337).

Thanks for making it future-proof!
